### PR TITLE
Replace README docs with RTD content

### PR DESCRIPTION
**Pull request by rhornung67**
# Replace README docs with RTD content

- This PR adds a User Guide and Developer Guide for the RAJA Perf Suite.

The new docs can be viewed here: https://rajaperf.readthedocs.io/en/task-new-docs/

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+Stream benchmark, the "Apps" group contains kernels extracted from
+real scientific computing applications, and so forth. Each kernel in the Suite 
+appears in multiple RAJA and non-RAJA (i.e., *baseline*) variants that use 
+parallel programming models supported by RAJA. Some kernels have multiple 
+tunings of a variant to explore the performance implications of options that 
+a programming model supports.
+
+.. note:: Available variants for a kernel do not need to include all possible
+          variants in the Suite. In some cases, a kernel appears only in the 
+          subset of variants that makes sense for the particular kernel.
```
Comment (MrBurmark) [inline docs/index.rst:49]: awkward

Comment (rhornung67) [reply docs/index.rst:49]: I think I cut and pasted that from the original README docs. I will push a fix.

File: docs/sphinx/user_guide/build.rst [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+The RAJA Performance Suite code is hosted on the 
+`GitHub RAJA Performance Suite project <https://github.com/LLNL/RAJAPerf>`_. 
+To get the code, clone the repository into a local working space using the 
+command::
+
+   $ git clone --recursive https://github.com/LLNL/RAJAPerf.git
+
+The ``--recursive`` option is required to pull all RAJA Performance Suite 
+Git *submodules* on which it depends into your local copy of the repository.
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/build.rst:49]: awkward

File: docs/sphinx/user_guide/build.rst [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+command, you can also type the following command in the ``RAJAPerf`` 
+directory after cloning::
+
+  $ git submodule update --init --recursive
+
+Either way, the result is the same and you should be good to configure the
+code and build it.
+
+.. note:: * If you are in your local copy of the RAJA Performance Suite repo
+            and you switch to a different repo branch (e.g., you run the
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/build.rst:70]:
```suggestion
            and you switch to a different branch (e.g., you run the 
```

File: docs/sphinx/user_guide/build.rst [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+code and build it.
+
+.. note:: * If you are in your local copy of the RAJA Performance Suite repo
+            and you switch to a different repo branch (e.g., you run the 
+            command ``git checkout <different branch name>``), you may need to 
+            run the command ``git submodule update`` to set the Git *submodule
+            versions* to what is used by the new branch. To see if this is 
+            required, the ``git status`` command will indicate whether the
+            submodules are at the proper versions. 
+          * If the *set of submodules* in a new branch is different than the
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/build.rst:76]: Is this bullet covered in the previous?

File: docs/sphinx/user_guide/build.rst (hunk 2 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/build.rst]: Fixed.

File: docs/sphinx/user_guide/build.rst (hunk 3 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/build.rst]: Changed.

File: docs/sphinx/user_guide/build.rst (hunk 4 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/build.rst:76]: It's actually different. Same submodules but different versions (first bullet) vs. different submodules (second bullet).

File: docs/sphinx/user_guide/build.rst [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+These scripts are designed to be run in the top-level RAJAPerf directory. Each 
+script creates a descriptively-named build space directory and runs CMake with 
+a configuration appropriate for the platform and specified compiler(s). To 
+compile the code after CMake completes, enter the build directory and type 
+``make`` (or ``make -j <N>`` or ``make -j`` for a parallel build using N 
+processor cores, or all available processor cores on a node, respectively). 
+For example::
+
+  $ ./scripts/lc-builds/blueos_nvcc_clang.sh 10.2.89 70 10.0.1
+  $ cd build_blueos_nvcc10.2.89-sm_70-clang10.0.1
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/build.rst:136]:
```suggestion
  $ cd build_blueos_nvcc10.2.89-70-clang10.0.1
```

Comment (rhornung67) [reply docs/sphinx/user_guide/build.rst:136]: Good catch. Fix coming.

File: docs/sphinx/user_guide/build.rst [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+will be built. If you want to build both the Suite tests and RAJA tests (to
+verify that everything is built properly), pass the following options to 
+CMake: ``-DENABLE_TESTS=On`` and ``-DRAJA_PERFSUITE_ENABLE_TESTS=On``. This 
+can be done on the command line if you run CMake directly or by editing the 
+build script you are using. If you want to build the Suite tests, but not 
+RAJA tests, pass the two CMake options above plus the option 
+``-DRAJA_ENABLE_TESTS=Off``. In any case, after the build completes, you can 
+type ``make test`` to run the tests you have built and see the results.
+
+.. note:: Which kernel variants that can be run depends on which programming
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/build.rst:157]:
```suggestion
.. note:: The kernel variants that can be run depends on which programming
```

File: docs/sphinx/user_guide/build.rst [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+          model features have been enabled in a build configuration. By 
+          default, only *sequential* CPU RAJA and baseline variants will be 
+          built. To additionally enable OpenMP variants, for example, you must 
+          pass the ``DENABLE_OPENMP=On`` option to CMake. Similar options will
+          enable other variants for CUDA, HIP, and other programming models.
+
+.. note:: For GPU-enabled builds, only one GPU back-end can be enabled in a 
+          single executable. However, CPU and GPU enabled execution can be
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/build.rst:165]:
```suggestion
          single executable. However, CPU and GPU execution can be 
```

File: docs/sphinx/user_guide/build.rst [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+
+Building with MPI
+-----------------
+
+Earlier, we mentioned that the Suite can be built with MPI enabled and
+described why this is useful. Some configuration scripts we provide will 
+configure a build with MPI support enabled. For example::
+
+  $ ./scripts/lc-builds/lc-blueos_spectrum_nvcc_clang.sh rolling-release 10.2.89 70 10.0.1
+  $ cd build_lc_blueos-spectrumrolling-release-nvcc10.2.89-sm_70-clang10.0.1
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/build.rst:178]:
```suggestion
  $ cd build_lc_blueos-spectrumrolling-release-nvcc10.2.89-70-clang10.0.1
```

File: docs/sphinx/user_guide/run.rst [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+about kernel and execution details will also appear in a run report files 
+generated in the run directory after Suite execution completes. 
+
+.. note:: You can pass the ``--dryrun`` command-line option to the executable to see a summary of how the Suite will execute without actually running it.
+
+The Suite can be run in a variety of ways determined by the command-line 
+options passed to the executable. For example, you can run or exclude subsets 
+of kernels, variants, or groups. You can also pass options to set problem 
+sizes, number of times each kernel is run (sampled), and many other run 
+parameters.The goal is to build the code once and use scripts or other means
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/run.rst:41]:
```suggestion
parameters. The goal is to build the code once and use scripts or other means 
```

File: docs/sphinx/user_guide/build.rst (hunk 6 above)
Comment (rhornung67) [reply docs/sphinx/user_guide/build.rst:157]: I'll push changes based on your review comments when you are done with each file.

File: docs/sphinx/user_guide/run.rst [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+generated in the run directory after Suite execution completes. 
+
+.. note:: You can pass the ``--dryrun`` command-line option to the executable to see a summary of how the Suite will execute without actually running it.
+
+The Suite can be run in a variety of ways determined by the command-line 
+options passed to the executable. For example, you can run or exclude subsets 
+of kernels, variants, or groups. You can also pass options to set problem 
+sizes, number of times each kernel is run (sampled), and many other run 
+parameters.The goal is to build the code once and use scripts or other means 
+to run the Suite in different ways for analyses you want to perform.
```
Comment (MrBurmark) [inline docs/sphinx/user_guide/run.rst:42]:
```suggestion
to run the Suite in different ways for the analyses you want to perform.
```

**Comment by rhornung67** [comment]
@MrBurmark I believe I addressed all of your review comments.  @artv3 did you want to review this?

**Review by artv3 — APPROVED** [review]

**Comment by rhornung67** [comment]
@MrBurmark any more comments or suggestions?

**Pull request merged**